src/probnum/problems/zoo/quad/_quadproblems_uniform.py

```diff
@@ -0,0 +1,802 @@
+    assert len(a.shape) == 1 and a.shape[0] == dim
+    assert np.all(u <= 1.0) and np.all(u >= 0.0)
+
+    # Check that the input points have valid values
+    assert np.all(x <= 1.0) and np.all(x >= 0.0)
+
+    # Compute function values
+    f = np.cos(2.0 * np.pi * u[0] + np.sum(a * x, axis=1))
+
+    return f.reshape((n, 1))
+
+
+def Genz_productpeak(
+    x: np.ndarray, a: np.ndarray = None, u: np.ndarray = None
+) -> np.ndarray:
+    """Genz 'Product Peak' test function on [0,1]^d.
+
+    .. math::  f(x) = \prod_{i=1}^d ( a_i^{-2} + (x_i-u_i)^2)^{-1}.
+
+
+    Parameters
+    ----------
+        x
+            Array of points at which to evaluate the function.
+            Each row is a point and column corresponds to dimensions.
+            All entries should be in [0,1].
+        a
+            First set of parameters affecting the difficulty of the integration problem.
+        u
+            Second set of parameters affecting the difficulty of the integration problem.
+            All entries should be in [0,1].
+
+    Returns
+    -------
+        f
+            array of function evaluations at points 'x'.
+
+    References
+    ----------
+        https://www.sfu.ca/~ssurjano/prpeak.html
+    """
+
+    (n, dim) = x.shape
+
+    # Specify default values of parameters a and u
+    if a is None:
+        a = np.repeat(5.0, dim)
+    if u is None:
+        u = np.repeat(0.5, dim)
+
+    # Check that the parameters have valid values and shape
+    assert len(u.shape) == 1 and u.shape[0] == dim
+    assert len(a.shape) == 1 and a.shape[0] == dim
+    assert np.all(u <= 1.0) and np.all(u >= 0.0)
+
+    # Check that the input points have valid values
+    assert np.all(x <= 1.0) and np.all(x >= 0.0)
+
+    # Reshape u into an (n,dim) array with identical rows
+    u = np.repeat(u.reshape([1, dim]), n, axis=0)
+
+    # Compute function values
+    f = np.prod((a ** (-2) + (x - u) ** 2) ** (-1), axis=1)
+
+    return f.reshape((n, 1))
+
+
+def Bratley1992(x: np.ndarray) -> np.ndarray:
+    """'Bratley 1992' test function on [0,1]^d.
+
+    .. math::  f(x) = \sum_{i=1}^d (-1)^i \prod_{j=1}^i x_j.
+
+
+    https://www.sfu.ca/~ssurjano/bratleyetal92.html
+
+    Parameters
+    ----------
+        x
+            Array of points at which to evaluate the function.
+            All entries should be in [0,1].
+
+    Returns
+    -------
+        f
+            array of function evaluations at points 'x'.
+
+    References
+    ----------
+        Bratley, P., Fox, B. L., & Niederreiter, H. (1992). Implementation and tests of low-discrepancy sequences. ACM Transactions on Modeling and Computer Simulation (TOMACS), 2(3), 195-213.
+    """
+
+    (n, dim) = x.shape
+
+    # Check that the input points have valid values
+    assert np.all(x <= 1.0) and np.all(x >= 0.0)
+
+    # Compute function values
+    f = np.zeros(n)
+    for i in range(1, dim + 1):
+        f = f + ((-1.0) ** i) * np.prod(x[:, range(i)], axis=1)
```

Comment:
This should be faster:

```suggestion
    f = np.sum(
        ((-1.0) ** np.arange(1, dim + 1)) * np.cumprod(x, axis=1),
        axis=1,
    )
```